pkg/storegateway/storepb/indexheader.proto

```diff
@@ -0,0 +1,25 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+syntax = "proto3";
+
+package thanos;
+
+option go_package = "storepb";
+
+import "github.com/gogo/protobuf/gogoproto/gogo.proto";
+
+option (gogoproto.sizer_all) = true;
+option (gogoproto.marshaler_all) = true;
+option (gogoproto.unmarshaler_all) = true;
+option (gogoproto.goproto_getters_all) = false;
+
+// Do not generate XXX fields to reduce memory footprint.
+option (gogoproto.goproto_unkeyed_all) = false;
+option (gogoproto.goproto_unrecognized_all) = false;
+option (gogoproto.goproto_sizecache_all) = false;
+
+message HeadersLazyLoadedTrackerState {
+  map<string, int64> lazy_loaded_blocks = 1;
+  string user_id = 2;
+  int32 checksum = 3;
```

Comment:
This checksum doesn't seem to be populated anywhere - should it be?

Reply:
I was planning to add that in a separate PR. But will see if can just add it here.

Reply:
If it's a separate PR, then that's fine, just wanted to make sure it hadn't been left in by mistake